hadoop-ozone/recon/src/main/java/org/apache/hadoop/ozone/recon/tasks/ScmTableCountTask.java

```diff
@@ -0,0 +1,217 @@
+      LOG.error("Error while performing table count: {}", t);
+    }
+  }
+
+  /**
+   * Processes the table count by iterating over SCM tables and retrieving the
+   * counts of objects in each table. The counts are then stored in the Recon
+   * database.
+   *
+   * @throws IOException if an I/O error occurs during table count processing.
+   */
+  public void processTableCount() throws IOException {
+    // Acquire write lock
+    lock.writeLock().lock();
+    try {
+      // Initialize the object count map
+      Map<String, Long> objectCountMap = initializeCountMap();
+
+      // Iterate over SCM tables
+      for (String tableName : getTaskTables()) {
+        Table table = scmDBStore.getTable(tableName);
+
+        try (TableIterator keyIter = table.iterator()) {
+          // Retrieve the count of objects in the table
+          long count = Iterators.size(keyIter);
+          objectCountMap.put(getRowKeyFromTable(tableName), count);
+        } catch (IOException ioEx) {
+          LOG.error("Unable to populate SCM Table Count in Recon DB.", ioEx);
+        }
+      }
+
+      // Write the counts to the Recon db if the object count map is not empty
+      if (!objectCountMap.isEmpty()) {
+        writeCountsToDB(objectCountMap);
+      }
+
+      objectCountMap.clear();
+      LOG.info("Completed writing SCM table counts to DB.");
+    } finally {
+      // Release write lock
+      lock.writeLock().unlock();
+    }
+  }
+
+
+  /**
+   * Writes the object counts from the object count map to the Recon database.
+   *
+   * @param objectCountMap a map containing table names as keys and their object
+   *                       counts as values.
+   */
+  private void writeCountsToDB(Map<String, Long> objectCountMap) {
+    List<ScmTableCount> insertToDb = new ArrayList<>();
+
+    // Iterate over the object count map
+    for (Map.Entry<String, Long> entry : objectCountMap.entrySet()) {
+      // Create a new ScmTableCount object
+      ScmTableCount scmTableCountRecord = new ScmTableCount();
+      scmTableCountRecord.setTableName(entry.getKey());
+      scmTableCountRecord.setCount(entry.getValue());
+
+      // Add the ScmTableCount object to the list
+      insertToDb.add(scmTableCountRecord);
+    }
+    // Insert the list of ScmTableCount objects into the Recon database
+    scmTableCountDao.insert(insertToDb);
+  }
+
+  private Map<String, Long> initializeCountMap() throws IOException {
+    Collection<String> tables = getTaskTables();
+    Map<String, Long> objectCountMap = new HashMap<>(tables.size());
+    for (String tableName: tables) {
+      String key = getRowKeyFromTable(tableName);
+      objectCountMap.put(key, 0L);
+    }
+    return objectCountMap;
+  }
+
+
+  /**
+   * Returns the list of SCM tables to be processed by the task.
+   *
+   * @return the list of SCM tables to be processed by the task.
+   * @throws IOException if an I/O error occurs.
+   */
+  public Collection<String> getTaskTables() throws IOException {
```

Comment:
This method cannot throw IOException. Please remove it.

Reply:
Sorry about that, I must have added it by mistake.
I'll make the changes thanks!